api/src/main/java/jakarta/data/repository/Query.java

```diff
@@ -94,30 +135,30 @@
 public @interface Query {
 
     /**
-     * <p>Defines the query to be executed when the annotated method is called.</p>
+     * <p>Specifies the query executed by the annotated repository method,
+     * in JDQL or JPQL.</p>
      *
-     * <p>If an application defines a repository method with {@code @Query}
-     * and supplies other forms of sorting (such as {@link Sort}) to that method,
-     * then it is the responsibility of the application to compose the query in
-     * such a way that an {@code ORDER BY} clause (or query language equivalent)
-     * can be validly appended. The Jakarta Data provider is not expected to
-     * parse query language that is provided by the application.</p>
+     * <p>If the annotated repository method accepts other forms of sorting
+     * (such as a parameter of type {@link Sort}), it is the responsibility
+     * of the application programmer to compose the query so that an
+     * {@code ORDER BY} clause can be validly appended to the text of the
+     * query.</p>
      *
      * @return the query to be executed when the annotated method is called.
      */
     String value();
 
     /**
-     * <p>Defines an additional query that counts the number of elements that are
-     * returned by the {@link #value() primary} query. This is used to compute
-     * the {@link Page#totalElements total elements} and {@link Page#totalPages
-     * total pages} for paginated repository queries which are annotated with
-     * {@code @Query} and return a {@link Page} or {@link CursoredPage}.</p>
+     * <p>Specifies an additional query that counts the number of elements
+     * returned by the {@linkplain #value() primary query} and is used
+     * to compute the {@linkplain Page#totalElements total elements} and
+     * {@linkplain Page#totalPages total pages} for paginated repository
+     * methods returning {@link Page} or {@link CursoredPage}.</p>
      *
-     * <p>The default value of empty string indicates that no counting query
-     * is provided. A counting query is unnecessary when pagination is
-     * performed with {@link jakarta.data.page.PageRequest#withoutTotal()} or
-     * when pagination is not used at all.</p>
+     * <p>The additional query is optional. It is not used when pagination
+     * is performed with {@link PageRequest#withoutTotal()} or when the
+     * repository method returns a type other than {@link Page} or
+     * {@link CursoredPage}.</p>
```

Comment:
Clarify when it is required,
```suggestion
     * <p>The additional query is optional in most cases. It is required when
     * the return type is {@link Page} or {@link CursoredPage} and
     * pagination is performed with {@link PageRequest#requestTotal()}
     * of {@code true} and the primary query includes a `SELECT` or
     * `ORDER BY` clause.</p>
```

Reply:
I would remove this requirement too. It's completely trivial for the Jakarta Data provider to replace everything before `from` with `select count(this)`.

Reply:
In fact, the separate `count()` query is just no longer needed. That's one advantage of having our own query language!